sdk/sdk-core/aar/src/main/java/mil/emp3/core/mapgridlines/AbstractMapGridLine.java

```diff
@@ -0,0 +1,376 @@
+
+    /**
+     * The sub class must override this method to set the properties of the path. If the grid objject type
+     * was not created by the sub class it must call the parent method.
+     * @param path              The path feature created by the sub class.
+     * @param gridObjectType    The object type provided by the sub class.
+     */
+    protected void setPathAttributes(Path path, String gridObjectType) {
+    }
+
+    /**
+     * The sub class must override this method to set the properties of the label it creates. If the
+     * object class is not created by the sub class it must call the parents method.
+     * @param label             The Text feature created by the sub class.
+     * @param gridObjectType    The object type indicated by the sub class.
+     */
+    protected void setLabelAttributes(Text label, String gridObjectType) {
+    }
+
+    /**
+     * This method creates a path feature with the positions list provided, then set the path attributes.
+     * Then it calls the setPathAttributes to allow the sub class to set the path's properties.
+     * @param positionList      The list of IGeoPositions.
+     * @param gridObjectType    A caller defined string to identify the type of path.
+     * @return The IFeature interface of the path created.
+     */
+    protected IFeature createPathFeature(List<IGeoPosition> positionList, String gridObjectType) {
+        Path path = new Path(positionList);
+
+        path.setAltitudeMode(IGeoAltitudeMode.AltitudeMode.CLAMP_TO_GROUND);
+        if (containsStrokeStyle(gridObjectType)) {
+            path.setStrokeStyle(getStrokeStyle(gridObjectType));
+        }
+        setPathAttributes(path, gridObjectType);
+
+        return path;
+    }
+
+    /**
+     * This method creates a Text feature at the position indicated. Once the Text feature is created
+     * it sets the label style. Then it calls the setLabelAttribute method so the caller can set the labels properties.
+     * @param position          The IGeoPosition to place the label.
+     * @param text              The text to display on the map.
+     * @param gridObjectType    A caller defined string to identify the type of label.
+     * @return The IFeature interface of the Text feature created.
+     */
+    protected IFeature createLabelFeature(IGeoPosition position, String text, String gridObjectType) {
+        Text label = new Text(text);
+
+        label.setPosition(position);
+        if (containsLabelStyle(gridObjectType)) {
+            label.setLabelStyle(getLabelStyle(gridObjectType));
+        }
+        label.setAltitudeMode(IGeoAltitudeMode.AltitudeMode.CLAMP_TO_GROUND);
+        setLabelAttributes(label, gridObjectType);
+
+        if (this.currentCamera.getHeading() != 0.0) {
+            double azimuth = label.getAzimuth() - this.currentCamera.getHeading();
+
+            azimuth = ((((azimuth + 180) % 360.0) + 360.0) % 360.0) - 180.0;
```

Comment:
This operation converts the Double remainder operation to a modulus operation. It ensures that all label remain aligned with the grid when the heading changes.